### PR TITLE
Improvements to motor mixer to increase thrust margin

### PR DESCRIPTION
**Pull request by georgehines**

### CODE REVIEW

File: libraries/AP_HAL_AVR_SITL/SITL_State.cpp [hunk 1]
```diff
@@ -547,10 +547,12 @@ void SITL_State::_simulator_output(bool synthetic_clock_mode)
 		_motors_on = ((control.pwm[2]-1500)/500.0f) != 0;
 	} else {
 		_motors_on = false;
-        // apply engine multiplier to first motor
+        // apply engine multiplier to a motor
         control.pwm[0] = ((control.pwm[0]-1000) * _sitl->engine_mul) + 1000;
         // run checks on each motor
 		for (i=0; i<4; i++) {
+            // apply engine multiplier to all motors
+            control.pwm[i] = ((control.pwm[i]-1000) * _sitl->engine_mul) + 1000;
```
Comment (georgehines) [inline libraries/AP_HAL_AVR_SITL/SITL_State.cpp]: This appears to take advantage of the fly_stability_patch SITL test, but in doing so does not test whatever that was originally meant to test. Were we using that test for anything?

File: libraries/AP_HAL_AVR_SITL/SITL_State.cpp [hunk 2]
```diff
@@ -547,8 +547,6 @@ void SITL_State::_simulator_output(bool synthetic_clock_mode)
 		_motors_on = ((control.pwm[2]-1500)/500.0f) != 0;
 	} else {
 		_motors_on = false;
-        // apply engine multiplier to a motor
-        control.pwm[0] = ((control.pwm[0]-1000) * _sitl->engine_mul) + 1000;
         // run checks on each motor
```
Comment (georgehines) [inline libraries/AP_HAL_AVR_SITL/SITL_State.cpp]: Re-adding comment to current diff. This takes advantage of fly_stability_test, thereby breaking that test's intended function. Were we using that test for anything?

**Comment by georgehines** [comment]
OK I don't know what's going on with these outdated diffs. But anyhow I'm just trying to flag a SITL change that might otherwise go unnoticed.

File: mk/PX4/ROMFS/defaults.parm [hunk 3]
```diff
@@ -74,7 +74,7 @@ MOT_THR_MIX_MIN, 0.150000
 MOT_THST_BAT_MAX, 16.800000
 MOT_THST_BAT_MIN, 12.000000
 MOT_THST_EXPO, 0.800000
-MOT_THST_MAX, 0.940000
+MOT_THST_MAX, 0.965000
```
Comment (georgehines) [inline mk/PX4/ROMFS/defaults.parm]: is this uniform across copters, or copter-specific?

**Comment by georgehines** [comment]
I think this looks great, but would like to hold off merge until we iterate once on my questions above.

File: mk/PX4/ROMFS/defaults.parm (hunk 3 above)
Comment (priseborough) [inline mk/PX4/ROMFS/defaults.parm]: It is a characteristics of the esc and consistent for the units I tested. Given the gross correction inputs for attitude, a small amount of saturation inside the esc itself is preferable to leaving some thrust on the table.

**Comment by georgehines** [comment]
@wsilva32 can you speak to the SITL issue? are we using that test / do we think we ever need that test?

**Comment by priseborough** [comment]
I'm happy for the SITL patches to be pulled out as they were only required for testing the functionality.

**Comment by georgehines** [comment]
great, thanks

**Comment by proficnc** [comment]
@paul.... Feel free to chat, but careful of any saturation.

**Pull request merged**